notify/silences.go

```diff
@@ -101,12 +101,9 @@ func (am *GrafanaAlertmanager) CreateSilence(ps *PostableSilence) (string, error
 		return "", fmt.Errorf("%s: %w", msg, ErrCreateSilenceBadPayload)
 	}
 
-	silenceID, err := am.silences.Set(sil)
+	silenceID, err := am.silences.Upsert(sil)
```

Comment:
I think we should create a new `UpsertSilence` -- This way we can have the individual mode in Grafana use whatever mechanism it requires.